.github/actions/execute-assert-arc-e2e/action.yaml

```diff
@@ -0,0 +1,151 @@
+name: 'Execute and Assert ARC E2E Test Action'
+description: 'Queue E2E test workflow and assert workflow run result to be succeed'
+
+inputs:
+  github-auth-token:
+    description: 'GitHub access token to queue workflow run'
+    required: true
+  github-repo-owner:
+    description: "The repository owner name that has the test workflow file, ex: actions"
+    required: true
+  github-repo-name:
+    description: "The repository name that has the test workflow file, ex: test"
+    required: true
+  github-workflow-file:
+    description: 'The file name of the workflow yaml, ex: test.yml'
+    required: true
+  test-arc-name:
+    description: 'The name of the configured gha-runner-scale-set'
+    required: true
+  test-arc-namespace:
+    description: 'The namespace of the configured gha-runner-scale-set'
+    required: true
+  test-arc-controller-namespace:
+    description: 'The namespace of the configured gha-runner-scale-set-controller'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Queue test workflow
+      shell: bash
+      id: queue_workflow
+      run: |
+        queue_time=`date +%FT%TZ`
+        echo "queue_time=$queue_time" >> $GITHUB_OUTPUT
+        curl -X POST https://api.github.com/repos/${{inputs.github-repo-owner}}/${{inputs.github-repo-name}}/actions/workflows/${{inputs.github-workflow-file}}/dispatches \
+        -H "Accept: application/vnd.github.v3+json" \
+        -H "Authorization: token ${{inputs.github-auth-token}}" \
+        -d '{"ref": "main", "inputs": { "arc_name": "${{inputs.test-arc-name}}" } }'
+
+    - name: Wait test workflow run to start
+      uses: actions/github-script@v6
+      id: query_workflow
+      with:
+        script: |
```

Comment:
nit: Can we add one or two liner explaining the logic behind this script? It'll be easier to understand what's happening at a glance instead of diving into the code